add information about Zaraz JSONata support (#8457)
cloudflare/cloudflare-docs: content/zaraz/advanced/using-jsonata.md

---
pcx_content_type: how-to
title: Using JSONata
weight: 10
---

# Using JSONata

> For advanced use cases, it is sometimes useful to be able to retrieve a value in a particular way. For instance, you might be using zaraz.track to send a list of products to Zaraz, but the third-party tool you want to send this data to requires the total cost of the products. Alternatively, you may want to manipulate a value, such as converting it to lowercase.

Reviewer: ```suggestion
For advanced use cases, it is sometimes useful to be able to retrieve a value in a particular way. For instance, you might be using `zaraz.track` to send a list of products to Zaraz, but the third-party tool you want to send this data to requires the total cost of the products. Alternatively, you may want to manipulate a value, such as converting it to lowercase.
```

Cloudflare Zaraz uses JSONata to enable you to perform complex operations on your data. With JSONata, you can evaluate expressions against the [Zaraz Context](/zaraz/reference/context/), allowing you to access and manipulate a wide range of values. To learn more about the values available and how to access them, consult the [full reference](/zaraz/reference/context/). You can also refer to the [complete JSONata documentation](https://docs.jsonata.org/) for more information about JSONata's capabilities.

> To use JSONata inside Zaraz, follow these easy steps:

Reviewer: ```suggestion
To use JSONata inside Zaraz, follow these steps:
```

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/), and select your account and domain.
2. Go to **Zaraz** > **Tools configuration** > **Tools**.

> 2. Click **Edit** next to a tool that you have already configured.

Reviewer: ```suggestion
3. Select **Edit** next to a tool that you have already configured.
```

> 3. Select an action or add a new one.

Reviewer: ```suggestion
4. Select an action or add a new one.
```

> 4. Choose the field you want to use JSONata in, and wrap your JSONata expression with double curly brackets, like `{{ expression }}`.

Reviewer: ```suggestion
5. Choose the field you want to use JSONata in, and wrap your JSONata expression with double curly brackets, like `{{ expression }}`.
```
